tests/unittests/tests-core/tests-core-clist.c

```diff
@@ -22,84 +22,198 @@ static list_node_t test_clist;
 
     list_node_t *elem = &(tests_clist_buf[1]);
 
-    test_clist_add_one();
+    test_clist_rpush();
 
-    clist_insert(list, elem);
+    clist_rpush(list, elem);
 
     TEST_ASSERT_NOT_NULL(list->next);
     TEST_ASSERT(list->next == elem);
     TEST_ASSERT(list->next->next->next == list->next);
 }
 
-static void test_clist_remove_head(void)
+static void test_clist_add_three(void)
+{
+    set_up();
+    list_node_t *list = &test_clist;
+
+    for (int i = 0; i < 3; i++) {
+        clist_rpush(list, &(tests_clist_buf[i]));
+    }
+
+    TEST_ASSERT_NOT_NULL(list->next);
+    TEST_ASSERT(list->next == &(tests_clist_buf[2]));
+    TEST_ASSERT(list->next->next == &(tests_clist_buf[0]));
+    TEST_ASSERT(list->next->next->next == &(tests_clist_buf[1]));
+    TEST_ASSERT(list->next->next->next->next == &(tests_clist_buf[2]));
+}
+
+static void test_clist_find(void)
+{
+    list_node_t *list = &test_clist;
+
+    test_clist_add_three();
+
+    for (int i = 0; i < 3; i++) {
+        TEST_ASSERT(clist_find(list, &(tests_clist_buf[i])) == &(tests_clist_buf[i]));
+    }
+
+    TEST_ASSERT_NULL(clist_find(list, &(tests_clist_buf[3])));
+}
+
+static void test_clist_find_before(void)
+{
+    list_node_t *list = &test_clist;
+
+    test_clist_add_three();
+
+    TEST_ASSERT(clist_find_before(list, &(tests_clist_buf[0])) == &(tests_clist_buf[2]));
+
+    for (int i = 1; i < 3; i++) {
+        TEST_ASSERT(clist_find_before(list, &(tests_clist_buf[i])) == &(tests_clist_buf[i-1]));
+    }
+
+    TEST_ASSERT_NULL(clist_find_before(list, &(tests_clist_buf[3])));
+}
+
+static void test_clist_remove(void)
+{
+    list_node_t *list = &test_clist;
+
+
+    for (int i = 0; i < 3; i++) {
+        test_clist_add_three();
+        clist_remove(list, &(tests_clist_buf[i]));
+
+        for (int j = 0; j < 3; j++) {
+            if (i == j) {
+                TEST_ASSERT_NULL(clist_find(list, &(tests_clist_buf[j])));
+            }
+            else {
+                TEST_ASSERT(clist_find(list, &(tests_clist_buf[j])) == &(tests_clist_buf[j]));
+            }
+        }
+    }
+
+    /* list now contains 0, 1 */
+    TEST_ASSERT(list->next == &(tests_clist_buf[1]));
+    TEST_ASSERT(list->next->next == &(tests_clist_buf[0]));
+
+    clist_remove(list, &(tests_clist_buf[1]));
+    TEST_ASSERT(list->next == &(tests_clist_buf[0]));
+    TEST_ASSERT(list->next->next == &(tests_clist_buf[0]));
+
+    clist_remove(list, &(tests_clist_buf[0]));
+    TEST_ASSERT_NULL(list->next);
+}
+
+static void test_clist_lpop(void)
+{
+    list_node_t *list = &test_clist;
+
+    test_clist_add_three();
+
+    TEST_ASSERT(clist_lpop(list) == &tests_clist_buf[0]);
+    TEST_ASSERT_NOT_NULL(list->next);
+
+    TEST_ASSERT(clist_lpop(list) == &tests_clist_buf[1]);
+    TEST_ASSERT_NOT_NULL(list->next);
+
+    TEST_ASSERT(clist_lpop(list) == &tests_clist_buf[2]);
+    TEST_ASSERT_NULL(list->next);
+    TEST_ASSERT_NULL(clist_lpop(list));
+}
+
+static void test_clist_lpush(void)
+{
+    set_up();
```

Comment:
set_up() is called automatically before this function is called so why do you call it?